src/utils/fetch.js

```diff
@@ -7,6 +7,14 @@ import {
   getDeleteMessage
 } from '../constants/lang';
 
+function BadInputError(message) {
+  this.name = 'BadInputError';
+  this.message = message || 'Bad input';
+  this.stack = (new Error()).stack;
+}
+BadInputError.prototype = Object.create(Error.prototype);
+BadInputError.prototype.constructor = BadInputError;
+
```

Comment:
Could you use ES6 class here and separate it into another file inside `utils` directory ?